UploadDaemon/SymbolAnalysis/LineCoverageSynthesizer.cs

```diff
@@ -15,12 +15,17 @@ public class LineCoverageSynthesizer : ILineCoverageSynthesizer
     {
         private static readonly Logger logger = LogManager.GetCurrentClassLogger();
 
-        private readonly IDictionary<string, IDictionary<GlobPatternList, SymbolCollection>> symbolCollectionsCache = new Dictionary<string, IDictionary<GlobPatternList, SymbolCollection>>();
+        private readonly SymbolCollectionResolver symbolCollectionResolver;
+
+        public LineCoverageSynthesizer(SymbolCollectionResolver symbolCollectionResolver)
```

Comment:
it seems that at every location where the `LineCoverageSynthesizer` is instantiated, you simply pass `new SymbolCollectionResolver()` here. If there's no reason at the moment to pass anything else, I'd say you can inline this in the field definiiton and thus make constructing this class a bit simpler.